fhir_zulip_nlp/fhir_zulip_nlp.py

```diff
@@ -206,41 +216,56 @@ def create_report2(
             df_kw = df_kw.sort_values(['timestamp'])  # oldest first
             threads: List[str] = list(df_kw['subject'].unique())
             # Average thread length
+            # TODO: Refactor to pandas?
             tot_thread_len = 0
-            for thread in range(len(threads)):
-                df_thread = df_kw[df_kw['subject'] == threads[thread]]
+            thread_data: Dict[str, pd.DataFrame] = {}
+            for thread in threads:
+                df_thread = df_kw[df_kw['subject'] == thread]
+                # TODO: Want to double check that timestamps are still/indeed sorted properly (i) here, and
+                #  (ii) everywhere else where we're doing timestamps like this
+                # TODO: better: rather than get the first and the last, timestamp. should be able to get max() and min()
                 z = (list(df_thread['timestamp'])[-1] - list(df_thread['timestamp'])[0]) / seconds_per_day
                 tot_thread_len += float(f'{z:.1f}')
+                thread_data[thread] = df_thread
             avg_thread_len = round(tot_thread_len / len(threads), 3)
             # Outliers
+            # TODO: Refactor to pandas to reduce lines and improve performance?
+            # TODO: Add cols for 1 and 2 std deviations?
+            # TODO: Might need to make 3 columns for these outliers: std deviations away from (i) keyword avg,
+            #  (ii) category avg, (iii) avg of all of our queried category/keyword threads.
             tot = 0
             stddevthread = 0
-            for x in range(len(threads)):
-                df_thread1 = df_kw[df_kw['subject'] == threads[x]]
-                z1 = (list(df_thread1['timestamp'])[-1] - list(df_thread1['timestamp'])[0]) / seconds_per_day
+            for thread in threads:
```

Comment:
@rohaher idk if you changed something, but i was able to change the code to iterate over thread names rather than the index w/out causing any additional issues.